### PR TITLE
Bug 1271989 - error when navigating to resources in diff projects

### PR DESCRIPTION
**Pull request by jwforres**
Also fixes an issue with the project dropdown not containing all projects
when on a page that uses the new project service instead of the nested controllers

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett review

File: assets/app/scripts/services/data.js [hunk 1]
```diff
@@ -704,7 +704,10 @@ angular.module('openshiftConsole')
   DataService.prototype._uniqueKeyForResourceContext = function(resource, context) {
     // Note: when we start handling selecting multiple projects this
     // will change to include all relevant scope
-    if (context.project && context.project.metadata) {
+    if (resource === "projects") { // when we are loading the projects list we dont need additional context
```
Comment (spadgett) [inline assets/app/scripts/services/data.js]: Are there other resources that don't need a namespace? `projectrequests`?

/cc @liggitt

Comment (jwforres) [inline assets/app/scripts/services/data.js]: do we get/list/watch projectrequests?  uniqueKey is only used for that

Comment (spadgett) [inline assets/app/scripts/services/data.js]: We do a GET on it in projects.js.

File: assets/app/scripts/services/data.js [hunk 2]
```diff
@@ -704,7 +704,10 @@ angular.module('openshiftConsole')
...
+      return resource;
+    }
+    else if (context.project && context.project.metadata) {
```
Comment (liggitt) [inline assets/app/scripts/services/data.js]: I think `context.namespace` should take priority... can you debug the places we look up things from the shared namespace and make sure their cache key is correct?

**Comment by jwforres** [comment]
[merge]

**Pull request merged**